crates/zkvm-jetpack/src/jets/verifier_jets.rs

```diff
@@ -176,3 +202,228 @@ fn process_belt(
 
     (acc, num)
 }
+
+// Combined roll on tap by
+fn roll_by<T, F: FnMut(T, Noun, Noun) -> Result<T, JetErr>>(
```

Comment:
This is a rather complicated function to test correctness of.

Reply:
I added a comment explaining the code better, I don't see how that can have different semantics than `(roll (tap by ...` (besiders, the order for accumulator update is irrelevant for this function:

```
Ok(acc + (coeff * res))
```

However, if you'd like more assurance, I can rewrite this to use `HoonMapIter`+fold, as that's equivalent logic split into 2 parts.